test/runtests.jl

```diff
@@ -717,6 +717,8 @@ end
     @test @inferred(ArrayInterface.strides(u_view)) == (4,)
     @test @inferred(ArrayInterface.strides(u_view_reinterpreted)) == (4,)
     @test @inferred(ArrayInterface.strides(u_view_reshaped)) == (4, 4)
+
+    @test_broken @inferred(ArrayInterface.axes(u_vectors)) isa ArrayInterface.axes_types(u_vectors)
```

Comment:
Does inference fail here?

Reply:
No, it's that it isn't of the type returned by `axes_types`.

Reply:
I need to do some clean up on `axes` and can fix this. I think we need to just go all in on `OptionallyStaticUnitRange` here.

Reply:
No worries, and no hurry (there's nothing wrong with a broken test, it's essentially a TODO list).

Reply:
Did you want me to fix this before this PR went forward or afterwards. Just want to make sure this PR isn't stalled because of me.